protocols/imap/src/main/java/org/apache/james/imap/processor/AuthenticateProcessor.java

```diff
@@ -184,14 +188,20 @@ private void doOAuth(String initialResponse, ImapSession session, ImapRequest re
         } else {
             OIDCSASLParser.parse(initialResponse)
                 .flatMap(oidcInitialResponseValue -> session.oidcSaslConfiguration()
-                    .flatMap(configuration -> new OidcJwtTokenVerifier().verifyAndExtractClaim(oidcInitialResponseValue.getToken(), configuration.getJwksURL(), configuration.getClaim())))
-                .flatMap(this::extractUserFromClaim)
+                    .flatMap(configure -> validateToken(configure, oidcInitialResponseValue.getToken())))
                 .ifPresentOrElse(username -> authSuccess(username, session, request, responder),
                     () -> manageFailureCount(session, request, responder, HumanReadableText.AUTHENTICATION_FAILED));
         }
         session.stopDetectingCommandInjection();
     }
 
+    private Optional<Username> validateToken(OidcSASLConfiguration oidcSASLConfiguration, String token) {
+        return Mono.just(new OidcJwtTokenVerifier(oidcSASLConfiguration.getIntrospectionEndpoint().map(DefaultIntrospectionClient::new)))
```

Comment:
The introspection client should be a field, injected. The URL should likely be a part of the method.